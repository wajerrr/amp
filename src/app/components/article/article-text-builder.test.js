import renderer from 'react-test-renderer';
import buildArticleText from './article-text-builder';

describe('buildComponents', () => {
  it('should match snapshot for tag a', () => {
    const tree = renderer
      .create(
        buildArticleText([
          {
            type: 'tag',
            name: 'a',
            attribs: {
              href: 'https://www.economist.com/',
            },
            children: [
              {
                data: 'Chain reaction',
                type: 'text',
              },
            ],
          },
        ])
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot for image', () => {
    const tree = renderer
      .create(
        buildArticleText([
          {
            type: 'tag',
            name: 'figure',
            attribs: {
              itemtype: 'https://schema.org/ImageObject',
            },
            children: [
              {
                type: 'tag',
                name: 'img',
                attribs: {
                  src:
                    'src="https://www.economist.com/sites/default/files/20180602_blp904.jpg"',
                  width: 1280,
                  height: 720,
                  alt: '',
                },
                children: [],
              },
            ],
          },
        ])
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot for iframe', () => {
    const tree = renderer
      .create(
        buildArticleText([
          {
            type: 'tag',
            name: 'figure',
            attribs: {
              itemtype: 'https://schema.org/MediaObject',
              class: 'op-interactive',
            },
            children: [
              {
                type: 'tag',
                name: 'iframe',
                attribs: {
                  height: '630',
                  src: '//infographics.economist.com/2017/ukelmap-2017/',
                  width: '640',
                },
                children: [],
              },
            ],
          },
        ])
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot for br', () => {
    const tree = renderer
      .create(
        buildArticleText([
          {
            type: 'tag',
            name: 'br',
            attribs: {},
            children: [],
          },
        ])
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
