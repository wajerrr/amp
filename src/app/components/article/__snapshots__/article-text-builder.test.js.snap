// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildComponents should match snapshot for br 1`] = `<br />`;

exports[`buildComponents should match snapshot for iframe 1`] = `
<figure>
  <amp-iframe
    frameborder="0"
    height={630}
    layout="fixed-height"
    sandbox="allow-scripts allow-same-origin"
    src="https://infographics.economist.com/2017/ukelmap-2017/"
  >
    <div
      className="css-boochl-StyledPlaceholderImage eh37php0"
      height={86}
      width={172}
    >
      <amp-img
        height={86}
        layout="fixed"
        placeholder={true}
        src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIj8+PHN2ZyB3aWR0aD0iMTcyIiBoZWlnaHQ9Ijg2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnN2Zz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPiA8c3R5bGU+LnN0MHtmaWxsOiNGMkYyRjI7fSAuc3Qxe2ZpbGw6I0NDQ0NDQzt9IC5zdDJ7ZmlsbDojRkZGRkZGO308L3N0eWxlPiA8Zz4gIDx0aXRsZT5MYXllciAxPC90aXRsZT4gIDxwYXRoIGlkPSJzdmdfMiIgZD0ibTAsMGwxNzIsMGwwLDg2bC0xNzIsMGwwLC04NnoiIGNsYXNzPSJzdDEiIGZpbGw9ImJsYWNrIi8+ICA8cGF0aCBpZD0ic3ZnXzMiIGQ9Im03OS44LDIyLjdjLTAuMSwtMi40IC0wLjYsLTIuNyAtNC4yLC0yLjdsLTAuOCwwbDAsMTMuOWMwLDEuOSAwLjUsMi40IDIuNCwyLjVsMC4xLDAuN2wtOC4xLDBsMC4xLC0wLjdjMS45LC0wLjEgMi40LC0wLjYgMi40LC0yLjVsMCwtMTMuOWwtMC45LDBjLTMuOCwwIC00LjMsMC4zIC00LjMsMi43bC0wLjksLTAuMWwwLjEsLTMuOWMzLjUsMCA2LDAuMSA3LjgsMC4xYzEuNywwIDQuMiwwIDcuNSwtMC4xbDAuMSwzLjlsLTEuMywwLjF6bTEyLjksMTQuMmMtMC43LDAgLTEuOCwwLjEgLTMuMiwwLjFsMC4xLC0wLjZjMS4zLC0wLjIgMS41LC0wLjUgMS41LC0yLjdsMCwtMy45YzAsLTMuMiAtMC42LC00LjMgLTIuMiwtNC4zYy0wLjksMCAtMS45LDAuNiAtMi4zLDEuNWMtMC4yLDAuMyAtMC4zLDAuNiAtMC40LDFsMCw1LjdjMCwyLjMgMC4yLDIuNSAxLjUsMi43bDAuMSwwLjZjLTEuNCwtMC4xIC0yLjUsLTAuMSAtMy4yLC0wLjFjLTAuNywwIC0xLjksMC4xIC0zLjMsMC4xbDAuMSwtMC42YzEuNywwIDEuOSwtMC4zIDIsLTIuN2wwLC0xMS43YzAsLTIuNCAtMC4zLC0yLjcgLTIsLTIuN2wtMC4xLC0wLjZjMi4zLC0wLjEgMi44LC0wLjIgNC45LC0wLjZsMCw3LjljMS4zLC0xLjUgMi41LC0yLjEgNCwtMi4xYzIuNSwwIDMuOSwxLjYgMy45LDQuNWwwLDUuM2MwLDIuNCAwLjMsMi43IDIsMi43bDAuMSwwLjZjLTEuNywtMC4xIC0yLjgsLTAuMSAtMy41LC0wLjF6bTguMSwtNy4xYzAuMSwzLjkgMS41LDYgMy44LDZjMS40LDAgMi42LC0wLjYgMy44LC0yLjFsMC40LDAuN2MtMS42LDIgLTMuMSwyLjggLTUuMiwyLjhjLTMuNSwwIC01LjksLTIuNiAtNS45LC02LjRjMCwtNC4xIDIuNSwtNi45IDYsLTYuOWMyLDAgMy4zLDAuNyA0LjMsMi4zYzAuNiwxLjEgMC44LDEuOCAwLjgsMy42bC04LDB6bTIuNSwtNWMtMS41LDAgLTIuMywxLjMgLTIuNiw0YzIuMywtMC4xIDMsLTAuMSA1LjQsLTAuNGMtMC4zLC0yLjQgLTEuMiwtMy42IC0yLjgsLTMuNnptLTgwLjgsMzYuM2MzLjIsMCA0LjEsLTAuNyA0LjQsLTMuNGwxLjEsMC4xbC0wLjYsNC43bC0yLjIsMGwtNi45LC0wLjFsLTQuNywwbDAuMSwtMC43YzIuMSwtMC4xIDIuNiwtMC42IDIuNiwtMi43bDAsLTEzLjNjMCwtMi4xIC0wLjUsLTIuNyAtMi42LC0yLjdsLTAuMSwtMC43bDQuNCwwLjFsNi45LC0wLjFsMS44LDBsMC4xLDQuMWwtMS4xLDAuMWMtMC4yLC0yLjYgLTAuNywtMi44IC00LjcsLTIuOGwtMC45LDBsMCw3LjdsMS4yLDBjMC41LDAgMC45LDAuMSAxLjMsMC4xYzIuMywwIDIuOCwtMC41IDIuOCwtMi42bDEsMGMtMC4xLDEuNyAtMC4xLDIuNiAtMC4xLDMuNGwwLDIuOWwtMC45LDBjLTAuMSwtMiAtMC42LC0yLjQgLTIuNywtMi40bC0xLjUsMGwtMS4xLDAuMWwwLDUuOWMtMC4xLDIuMiAwLDIuMyAyLjQsMi4zem0xNS43LC0xMi4xYy0yLjEsMCAtMy4yLDEuOSAtMy4yLDUuNWMwLDQuMyAxLjMsNi43IDMuNyw2LjdjMS4yLDAgMi4zLC0wLjUgMy40LC0xLjdsMC40LDFjLTEuNiwxLjcgLTIuOSwyLjMgLTQuOCwyLjNjLTMuNywwIC02LjEsLTIuOSAtNi4xLC03LjNjMCwtNC42IDIuNiwtNy41IDYuNiwtNy41YzEuMSwwIDIuMiwwLjIgMy45LDAuNmwwLjEsMy40bC0xLjMsMC4xYy0wLjUsLTIuNSAtMSwtMy4xIC0yLjcsLTMuMXptMTQuMiwxMy44Yy00LDAgLTYuNywtMi43IC02LjksLTYuOGwwLC0wLjRjMCwtNC44IDIuNiwtNy41IDcsLTcuNWM0LjEsMCA3LDMgNyw3LjRjMCw0LjMgLTIuOCw3LjMgLTcuMSw3LjN6bS0wLjMsLTEzLjhjLTIuMSwwIC0zLjEsMS44IC0zLjEsNS4zYzAsMC40IDAsMC44IDAuMSwxLjNjMC4zLDQgMS42LDYuMiAzLjgsNi4yYzIsMCAzLjIsLTIuMSAzLjIsLTUuN2MwLC00LjEgLTEuNywtNy4xIC00LC03LjF6bTIwLDEzLjVsMC4xLC0wLjdjMS41LC0wLjIgMS43LC0wLjUgMS43LC0zbDAsLTQuM2MwLC0zLjYgLTAuNiwtNC43IC0yLjUsLTQuN2MtMS4xLDAgLTIsMC42IC0yLjYsMS42Yy0wLjIsMC40IC0wLjQsMS4xIC0wLjQsMS40bDAsNmMwLDIuNSAwLjIsMi44IDEuNywzbDAuMSwwLjdjLTEuNSwtMC4xIC01LjUsLTAuMSAtNy4yLDBsMC4xLC0wLjdjMS45LDAgMi4yLC0wLjQgMi4yLC0zbDAsLTYuNmMwLC0yLjYgLTAuMywtMyAtMi4yLC0zbC0wLjEsLTAuNmMyLjUsLTAuMSAzLjEsLTAuMiA1LjIsLTAuN2wwLjEsMi41YzEuNiwtMS44IDIuOCwtMi40IDQuNiwtMi40YzIuNywwIDQuMywxLjggNC4zLDVsMCw2LjVjMCwxLjggMC41LDIuMyAyLjIsMi4zbDAuMSwwLjdjLTEuOSwtMC4xIC01LjksLTAuMSAtNy40LDB6bTE2LjUsMC4zYy00LDAgLTYuNywtMi43IC02LjksLTYuOGwwLC0wLjRjMCwtNC44IDIuNiwtNy41IDcsLTcuNWM0LjEsMCA3LDMgNyw3LjRjMCw0LjMgLTIuOCw3LjMgLTcuMSw3LjN6bS0wLjMsLTEzLjhjLTIuMSwwIC0zLjEsMS44IC0zLjEsNS4zYzAsMC40IDAsMC44IDAuMSwxLjNjMC4zLDQgMS42LDYuMiAzLjgsNi4yYzIsMCAzLjIsLTIuMSAzLjIsLTUuN2MwLC00LjEgLTEuNywtNy4xIC00LC03LjF6bTI4LjUsMTMuNWwwLjEsLTAuN2MxLjUsLTAuMiAxLjYsLTAuNSAxLjcsLTNsMCwtNC4zYzAsLTMuNSAtMC42LC00LjcgLTIuNCwtNC43Yy0xLDAgLTIuMSwwLjcgLTIuNiwxLjZjLTAuMSwwLjMgLTAuMywwLjcgLTAuNCwxLjFjMCwwLjEgLTAuMSwwLjMgLTAuMSwwLjNsMCw2YzAsMi41IDAuMiwyLjggMS43LDNsMC4xLDAuN2MtMS41LC0wLjEgLTUuMiwtMC4xIC02LjcsMGwwLjEsLTAuN2MxLjUsLTAuMiAxLjYsLTAuNSAxLjcsLTNsMCwtNC4zYzAsLTMuNSAtMC42LC00LjcgLTIuNCwtNC43Yy0xLjEsMCAtMiwwLjYgLTIuNiwxLjZjLTAuMiwwLjUgLTAuNCwxLjEgLTAuNCwxLjRsMCw2YzAsMi41IDAuMiwyLjggMS43LDNsMC4xLDAuN2MtMS41LC0wLjEgLTUuNSwtMC4xIC03LjIsMGwwLjEsLTAuN2MxLjksMCAyLjIsLTAuNCAyLjIsLTNsMCwtNi42YzAsLTIuNiAtMC4zLC0zIC0yLjIsLTNsLTAuMSwtMC42YzIuNSwtMC4xIDMuMSwtMC4yIDUuMiwtMC43bDAuMSwyLjVjMS42LC0xLjggMi44LC0yLjQgNC41LC0yLjRjMS45LDAgMywwLjggMy45LDIuN2MxLjcsLTIgMywtMi43IDQuNywtMi43YzIuNywwIDQuMiwxLjggNC4yLDVsMCw2LjVjMCwxLjggMC41LDIuMyAyLjIsMi4zbDAuMSwwLjdjLTEuOSwtMC4xIC01LjksLTAuMSAtNy4zLDB6bTEzLjcsLTAuMmMtMC44LDAgLTIuMSwwLjEgLTMuOCwwLjFsMC4xLC0wLjdjMS45LDAgMi4yLC0wLjQgMi4yLC0zbDAsLTYuNmMwLC0yLjYgLTAuMywtMyAtMi4yLC0zbC0wLjEsLTAuNmMyLjYsLTAuMiAzLjIsLTAuMyA1LjUsLTAuN2wwLDEwLjljMCwyLjYgMC4zLDMgMi4yLDNsMC4xLDAuN2MtMS45LDAgLTMuMiwtMC4xIC00LC0wLjF6bS0wLjEsLTE3LjJjLTEuMSwwIC0xLjksLTAuOCAtMS45LC0xLjljMCwtMS4xIDAuOCwtMS45IDEuOSwtMS45YzEsMCAxLjksMC45IDEuOSwxLjljLTAuMSwxLjEgLTAuOSwxLjkgLTEuOSwxLjl6bTEyLDMuOWMtMS4zLDAgLTIuMiwwLjcgLTIuMiwxLjhjMCwxIDAuNSwxLjYgMS45LDIuNWMxLjMsMC45IDIuNCwxLjcgMy4xLDIuM2MwLjgsMC44IDEuMiwxLjcgMS4yLDIuOGMwLDIuNSAtMi4xLDQuNCAtNC44LDQuNGMtMSwwIC0yLC0wLjIgLTMuOCwtMC44bC0wLjEsLTIuOWwxLjQsLTAuMWMwLjYsMi4yIDEuMSwyLjYgMi42LDIuNmMxLjMsMCAyLjEsLTAuNyAyLjEsLTEuN3MtMC42LC0xLjcgLTIuNCwtMi45Yy0wLjksLTAuNiAtMS41LC0xIC0xLjksLTEuM2MtMC43LC0wLjYgLTEuMywtMS4yIC0xLjUsLTJjLTAuMiwtMC40IC0wLjMsLTAuOSAtMC4zLC0xLjVjMCwtMi42IDEuNywtNC4yIDQuNCwtNC4yYzEsMCAxLjUsMC4xIDMuOCwwLjZsMC4xLDIuOGwtMS40LDAuMWMtMC4zLC0xLjkgLTAuOSwtMi41IC0yLjIsLTIuNXptMTQuNCwxMi4zYzAuNSwwIDAuOSwtMC4xIDEuNSwtMC4zbDAsMC44Yy0xLDAuNSAtMi4zLDAuOCAtMy42LDAuOGMtMS4yLDAgLTIuNCwtMC43IC0yLjksLTEuN2MtMC40LC0wLjkgLTAuNiwtMS43IC0wLjYsLTMuN2wwLC03LjJsLTEuOCwwbDAsLTEuM2MyLjIsLTAuMiAzLjEsLTEuMSAzLjcsLTQuMWwxLjMsMGwwLDMuOGwzLjUsMGwtMC40LDEuNmwtMy4yLDBsMCw2LjljMC4yLDMuNiAwLjYsNC40IDIuNSw0LjR6IiBjbGFzcz0ic3QyIiBmaWxsPSJibGFjayIvPiA8L2c+PC9zdmc+"
        width={172}
      />
    </div>
  </amp-iframe>
</figure>
`;

exports[`buildComponents should match snapshot for image 1`] = `
<figure>
  <amp-img
    alt=""
    height={720}
    layout="responsive"
    sizes="(min-width: 1000px) calc(1000px -  2 * 0.5em), calc(100vw - 2 * 0.5em)"
    src="src=\\"https://www.economist.com/sites/default/files/20180602_blp904.jpg\\""
    srcset="https://www.economist.com/sites/default/files/imagecache/200-width/sites/default/files/20180602_blp904.jpg\\" 200w,
https://www.economist.com/sites/default/files/imagecache/300-width/sites/default/files/20180602_blp904.jpg\\" 300w,
https://www.economist.com/sites/default/files/imagecache/400-width/sites/default/files/20180602_blp904.jpg\\" 400w,
https://www.economist.com/sites/default/files/imagecache/640-width/sites/default/files/20180602_blp904.jpg\\" 640w,
https://www.economist.com/sites/default/files/imagecache/800-width/sites/default/files/20180602_blp904.jpg\\" 800w,
https://www.economist.com/sites/default/files/imagecache/1000-width/sites/default/files/20180602_blp904.jpg\\" 1000w,
https://www.economist.com/sites/default/files/imagecache/1200-width/sites/default/files/20180602_blp904.jpg\\" 1200w,
https://www.economist.com/sites/default/files/imagecache/1280-width/sites/default/files/20180602_blp904.jpg\\" 1280w,
https://www.economist.com/sites/default/files/imagecache/1600-width/sites/default/files/20180602_blp904.jpg\\" 1600w"
    width={1280}
  />
</figure>
`;

exports[`buildComponents should match snapshot for tag a 1`] = `
<a
  className="css-1h5t0lq-StyledLink e2cf1b50"
  href="https://www.economist.com/"
>
  Chain reaction
</a>
`;
